firmware/crypto.c

```diff
@@ -95,12 +95,18 @@ int sshMessageSign(HDNode *node, const uint8_t *message, size_t message_len, uin
 
 int gpgMessageSign(HDNode *node, const uint8_t *message, size_t message_len, uint8_t *signature)
 {
-	// GPG should sign a SHA256 digest of the original message.
-	if (message_len != 32) {
-		return 1;
-	}
 	signature[0] = 0; // prefix: pad with zero, so all signatures are 65 bytes
-	return hdnode_sign_digest(node, message, signature + 1, NULL, NULL);
+	const curve_info *ed25519_curve_info = get_curve_by_name(ED25519_NAME);
```

Comment:
I have used `get_curve_by_name()` since ed25519_info is not exported from [bip32 module](https://github.com/romanz/trezor-crypto/blob/master/bip32.c#L45).